particles/Tutorial/Javascript/Demo/source/MoveApplier.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+ /* global defineParticle */
+
+ defineParticle(({DomParticle, html}) => {
+
+  // TODO (heimlich@) tied games
+  return class extends DomParticle {
+    update({move, gameState}, {}) {
+      // If the move is reset or start, reset the gameState
+      if (move && (move.move == 'start' || move.move == 'reset')) {
+        const gs = {
+          board: ',,,,,,,,',
+          gameOver: false,
+          winnerAvatar: null,
+          currentPlayer: 1,
+          lastMove: '',
+          gameStarted: true
+        };
+        this.updateSingleton('gameState', gs);
+      }
+      if (move && gameState && gameState.lastMove != move.move) {
+        // Get the old gameState values to update.
+        let newBoard = gameState.board;
+        let newGameOver = gameState.gameOver;
+        let newWinnerAvatar = gameState.winnerAvatar;
+        let newCurrentPlayer = gameState.currentPlayer;
+
+        // Get the board and move in a usable state
+        const arr = gameState.board.split(`,`);
+        const mv = parseInt(move.move, 10) - 1;
+        
+        // If the move is valid
+        if (arr[mv] == ``) {
+
+          // Apply move
+          arr[mv] = move.playerAvatar;
+          newBoard = arr.join();
+
+          // Define all the possible winning sequences
+          const winningSequences = [
```

Comment:
bikeshed: WDYT about defining a const outside the particle block? It is somewhat unusual to see consts that are meant to be used throughout the lifetime of an object to be defined inside the method body. But maybe this is caused by lack of support by other languages? Thoughts?